packages/grid/src/vaadin-grid-mixin.js

```diff
@@ -855,6 +856,16 @@ export const GridMixin = (superClass) =>
       this._getItem(index, row);
     }
 
+    /** @private */
+    _updateFocusedItem(row) {
+      const focusedItemInRow = row.__virtualIndex === this._focusedItemIndex;
+      let isFocusedCell;
+      iterateChildren(row, (cell) => {
+        isFocusedCell = cell === this._focusedCell;
```

Comment:
I'm not sure if this comparison is safe to use with the virtual scrolling mechanism. If you scroll down and up again, is the cell in the focused row guaranteed to reuse the same cell instance that was previously stored as `_focusedCell`?

Maybe @tomivirkki can provide some insight?

Reply:
From some testing it seems that this is not the case:

https://github.com/user-attachments/assets/11f2fbee-162a-478d-af7d-ab90b39b01ec

Reply:
How did you do that? And did you get focus just with a click? With 2 finger scrolling, it works for me as expected and I lose focus as soon as I press the scrollbar's handle